lib/dependencies/ImportContextDependency.js

```diff
@@ -20,13 +20,11 @@ class ImportContextDependency extends ContextDependency {
 	}
 
 	getWarnings() {
-		if(!this.critical) {
-			return;
+		let warnings = super.getWarnings() || [];
```

Comment:
Could you move the `CriticalDependencyWarning` logic from `ImportContextDependency` `RequireContextDependency` and `RequireResolveContextDependency` into the base class `ContextDependency`?